app/components/project-resource-quota/component.js

```diff
@@ -0,0 +1,111 @@
+import {  get, set, observer } from '@ember/object';
+import Component from '@ember/component';
+import { convertToMillis } from 'shared/utils/util';
+import { parseSi } from 'shared/utils/parse-unit';
+import layout from './template';
+
+export default Component.extend({
+  layout,
+
+  limit:          null,
+  nsDefaultLimit: null,
+  editing:        null,
+
+  quotaArray: null,
+
+  init() {
+    this._super(...arguments);
+
+    this.initQuotaArray();
+  },
+
+  actions: {
+    addQuota() {
+      get(this, 'quotaArray').pushObject({
+        key:            '',
+        projectLimit:   '',
+        namespaceLimit: '',
+      });
+    },
+
+    removeQuota(quota){
+      get(this, 'quotaArray').removeObject(quota);
+    }
+  },
+
+  quotaDidChange: observer('quotaArray.@each.{key,projectLimit,namespaceLimit}', function() {
+    const limit = {};
+    const nsDefaultLimit = {};
+
+    (get(this, 'quotaArray') || []).forEach((quota) => {
+      if ( quota.key && (quota.projectLimit || quota.namespaceLimit) ) {
+        limit[quota.key] = this.convertToString(quota.key, quota.projectLimit);
+        nsDefaultLimit[quota.key] = this.convertToString(quota.key, quota.namespaceLimit);
+      }
+    });
+
+    let out = null;
+
+    if ( Object.keys(limit).length ) {
+      out = {
+        resourceQuota:                 { limit },
+        namespaceDefaultResourceQuota: { limit: nsDefaultLimit },
+      }
+    }
+
+    this.sendAction('changed', out);
+  }),
+
+  convertToString(key, value) {
+    if ( !value ) {
```

Comment:
I think this function would be more readable (esp if we add more conditions) as a switch statement.